src/brasil/gov/portal/tests/test_controlpanel_portal.py

```diff
@@ -27,78 +20,50 @@ def setUp(self):
+        roles = ['Manager', 'Site Administrator']
+        for r in roles:
+            with api.env.adopt_roles([r]):
+                configlets = self.controlpanel.enumConfiglets(group='Products')
+                configlets = [a['id'] for a in configlets]
+                self.assertIn('portal', configlets, 'configlet not listed for ' + r)
 
 
-class ControlPanelFunctionalTestCase(unittest.TestCase):
-    """Testes funcionais da configuração do Portal Padrão."""
+class RegistryTestCase(unittest.TestCase):
 
-    layer = FUNCTIONAL_TESTING
+    layer = INTEGRATION_TESTING
 
     def setUp(self):
-        """Configura testes funcionais antes de cada teste."""
         self.portal = self.layer['portal']
-        self.portal_url = self.portal.absolute_url()
-        self.browser = Browser(self.layer['app'])
-        self.login_browser()
-        import transaction
-        transaction.commit()
+        self.registry = getUtility(IRegistry)
+        self.settings = self.registry.forInterface(ISettingsPortal)
 
-    def login_browser(self):
-        """Autentica usuário de teste no browser"""
-        setRoles(self.portal, TEST_USER_ID, ['Site Administrator'])
-        self.browser.handleErrors = False
-        basic_auth = 'Basic {0}'.format(
-            '{0}:{1}'.format(TEST_USER_NAME, TEST_USER_PASSWORD)
-        )
-        self.browser.addHeader('Authorization', basic_auth)
+    def test_esconde_autor_record_in_registry(self):
+        """Verifica se o registro esconde_autor existe."""
+        self.assertTrue(hasattr(self.settings, 'esconde_autor'))
+        self.assertFalse(self.settings.esconde_autor)
 
-    def test_configuracao_portal(self):
-        """Testa a configuração do Portal Padrão."""
-        label_autor = u'Esconde autor'
-        label_data = u'Esconde data de publicação'.encode('utf8')
-        self.browser.open('{0}/{1}'.format(self.portal.absolute_url(),
-                                           '@@overview-controlpanel'))
-        self.browser.getLink('.gov.br: Portal Padrão').click()
-        self.assertFalse(self.browser.getControl(label_autor).selected,
-                         'Campo Esconde autor com valor default errado.')
-        self.assertFalse(self.browser.getControl(label_data).selected,
-                         'Campo Esconde data com valor default errado.')
-        self.browser.getControl(label_autor).selected = True
-        self.browser.getControl(label_data).selected = True
-        self.browser.getControl(name='form.buttons.save').click()
-        self.assertTrue(self.browser.getControl(label_autor).selected,
-                        'Esconde autor não foi salvo.')
-        self.assertTrue(self.browser.getControl(label_data).selected,
-                        'Esconde data não foi salvo.')
+    def test_esconde_data_record_in_registry(self):
+        """Verifica se o registro esconde_data existe."""
+        self.assertTrue(hasattr(self.settings, 'esconde_data'))
```

Comment:
Don't use hasattr https://docs.plone.org/4/en/develop/plone-coredev/style.html#concrete-rules

Reply:
> Don't use hasattr() - this swallows exceptions, use getattr(foo, 'bar', None) instead. The problem with swallowed exceptions is not just poor error reporting. This can also mask ConflictErrors, which indicate that something has gone wrong at the ZODB level!

I agree we must probably not use `hasattr()` in code, but this is a simple test that can't hide anything else, including `ConflictErrors`.